### PR TITLE
allow plotting of Any vectors and 3D plots

### LINKED ISSUE #2146: Cannot convert Float64 to series data for plotting
Plots suddently stopped working on a `Any` array such as 
```julia
a = Any[1.2, 2.3, 3.3, 4.7]
plot(a)
```
is this new behavior of Plots, or there are package conflicts.
Here is my system: Julia v1.1.1 on win10

  [c52e3926] Atom v0.9.1
  [fbb218c0] BSON v0.2.3
  [336ed68f] CSV v0.5.11
  [159f3aea] Cairo v0.6.0
  [324d7699] CategoricalArrays v0.5.5
  [da1fd8a2] CodeTracking v0.5.7
  [c3611d14] ColorVectorSpace v0.6.2
  [5ae59095] Colors v0.9.5
  [861a8166] Combinatorics v0.7.0
  [a81c6b42] Compose v0.7.3
  [717857b8] DSP v0.6.0
  [a93c6f00] DataFrames v0.19.2
  [864edb3b] DataStructures v0.17.0
  [39dd38d3] Dierckx v0.4.1
  [31c24e10] Distributions v0.21.1
  [becb17da] Feather v0.5.3
  [5789e2e9] FileIO v1.0.7
  [587475ba] Flux v0.8.3
  [186bb1d3] Fontconfig v0.2.0
  [aa1b3936] GraphIO v0.4.0
  [a2cc645c] GraphPlot v0.3.1
  [bd48cda9] GraphRecipes v0.4.0
  [19dc6840] HCubature v1.4.0
  [f67ccb44] HDF5 v0.12.0
  [7073ff75] IJulia v1.19.0
  [51556ac3] ImageDistances v0.2.4
  [6218d12a] ImageMagick v0.7.5
  [916415d5] Images v0.18.0
  [c601a237] Interact v0.10.2
  [a98d9a8b] Interpolations v0.12.2
  [4138dd39] JLD v0.9.1
  [033835bb] JLD2 v0.1.2
  [e5e0dc1b] Juno v0.7.2
  [093fc24a] LightGraphs v1.2.0
  [2fda8390] LsqFit v0.8.1
  [23992714] MAT v0.5.0
  [ee78f7c6] Makie v0.9.4
  [626554b9] MetaGraphs v0.6.4
  [6f286f6a] MultivariateStats v0.6.0
  [47be7bcc] ORCA v0.2.1
  [f0f68f2c] PlotlyJS v0.12.5
  [91a5bcdd] Plots v0.26.1
  [92933f4c] ProgressMeter v1.0.0
  [438e738f] PyCall v1.91.2
  [d330b81b] PyPlot v2.8.1
  [295af30f] Revise v2.1.6
  [2913bbd2] StatsBase v0.32.0
  [65254759] StatsMakie v0.0.6
  [f3b207a7] StatsPlots v0.10.2
  [bd369af6] Tables v0.2.11
  [1986cc42] Unitful v0.16.0
  [0f1e0344] WebIO v0.8.9
  [ddb6d928] YAML v0.3.2
  [37e2e46d] LinearAlgebra

### LINKED ISSUE #2160: regression in parametrically defined surface plots
The following worked in `v0.26.0` to plot a parametric surface, but now fails in `v0.26.1` and `v0.26.2`:

```
using Plots

X(theta,phi) = sin(phi) * cos(theta)
Y(theta,phi) = sin(phi) * sin(theta)
Z(theta,phi) = cos(phi)
thetas = range(0, stop=pi/4,   length=50)
phis   = range(0, stop=pi/2, length=50)

xs = [X(theta, phi) for theta in thetas, phi in phis]
ys = [Y(theta, phi) for theta in thetas, phi in phis]
zs = [Z(theta, phi) for theta in thetas, phi in phis]

#pyplot() or plotly() worked, though not gr()
surface(xs, ys, zs)
```

Now, there is an error which looks like there is some issue with the data processing pipeline. Under `pyplot` we have this error:

```
ERROR: PyError ($(Expr(:escape, :(ccall(#= /Users/verzani/.julia/packages/PyCall/ttONZ/src/pyfncall.jl:44 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'ValueError'>
ValueError('x and y arrays must consist of at least 3 unique points')
```

Under `gr()` we also see an error:

```
QH6022 qhull input error: 0'th dimension's new bounds [-0.5, 0.5] too wide for
existing bounds [ 0,  0]

While executing:  | qhull d Qt QbB Qz
Options selected for Qhull 2015.2 2016/01/18:
  run-id 1761809233  delaunay  Qtriangulate  QbBound-unit-box 0.5
  Qz-infinity-point  _pre-merge  _zero-centrum  Pgood
Error in Delaunay triangulation calculation
```

`GR` directly also has an error:

```
 ***   IDENTICAL DATA POINTS.
   NDP = 2500   IP1 =    1   IP2 =    2   XD=0   YD=0
 ERROR DETECTED IN ROUTINE   IDTANG.
```

However, using `PyPlot.plot_surface(xs,ys,zs)` does properly render the surface under `v0.26.2`

### PR DESCRIPTION
**Pull request by daschw**
Closes #2146 and closes #2160.

### CODE REVIEW

**Comment by daschw** [comment]
Now also closes #2160.

File: src/series.jl [hunk 1]
```diff
@@ -21,32 +21,44 @@ handlemissings(s::Surface) = Surface(handlemissings(s.surf))
...
 
 # vector of data points is a single series
-convertToAnyVector(v::AVec{<:DataPoint}) = Any[prepareSeriesData(v)]
+convertToAnyVector(v::AVec{<:DataPoint}, plotattributes) = Any[prepareSeriesData(v)]
 
 # list of things (maybe other vectors, functions, or something else)
-convertToAnyVector(v::AVec) = vcat((convertToAnyVector(vi) for vi in v)...)
+function convertToAnyVector(v::AVec, plotattributes)
+    if all(x -> isa(x, Number) || ismissing(x), v) || all(x -> isa(x, AbstractString) || ismissing(x), v)
+        convertToAnyVector(convert.(DataPoint, v), plotattributes)
```
Comment (yha) [inline src/series.jl]: `convert.(DataPoint, v)` is not always strongly typed: e.g. `convert.(Plots.DataPoint, Any[1,2.3,missing])` gives a vector of `Any`, so it creates an infinite recursion here.

**Comment by jverzani** [comment]
Thanks!!

File: src/series.jl (hunk 1 above)
Comment (yha) [reply src/series.jl]: Oh, was looking at an outdated version, and didn't notice this was already merged.

**Comment by yha** [comment]
Suggested fix for the infinite recursion now in a new PR #2165

**Pull request merged**